BUILD

```diff
@@ -1,241 +1,280 @@
+    "gcr.io/{PROJECT_ID}/base-debian9:{COMMIT_SHA}": "//base:base_root_amd64_debian9",
+    "gcr.io/{PROJECT_ID}/base-debian10:{COMMIT_SHA}": "//base:base_root_amd64_debian10",
+}
+
+BASE.update({
+    "gcr.io/{PROJECT_ID}/base:" + tag_base + "-" + arch: "//base:" + label + "_" + user + "_" + arch + "_debian9"
+    for arch in ARCHITECTURES
+    for (tag_base, label, user) in [
+        ("latest", "base", "root"),
+        ("nonroot", "base", "nonroot"),
+        ("debug", "debug", "root"),
+        ("debug-nonroot", "debug", "nonroot"),
+    ]
+})
+
+BASE.update({
+    "gcr.io/{PROJECT_ID}/base-" + distro + ":" + tag_base + "-" + arch: "//base:" + label + "_" + user + "_" + arch + "_" + distro
+    for arch in ARCHITECTURES
+    for (tag_base, label, user) in [
+        ("latest", "base", "root"),
+        ("nonroot", "base", "nonroot"),
+        ("debug", "debug", "root"),
+        ("debug-nonroot", "debug", "nonroot"),
+    ]
+    for distro in DISTROS
+})
+
+CC = {
+    "gcr.io/{PROJECT_ID}/cc:{COMMIT_SHA}": "//cc:cc_root_amd64_debian9",
+    "gcr.io/{PROJECT_ID}/cc-debian9:{COMMIT_SHA}": "//cc:cc_root_amd64_debian9",
+    "gcr.io/{PROJECT_ID}/cc-debian10:{COMMIT_SHA}": "//cc:cc_root_amd64_debian10",
+}
+
+CC.update({
+    "gcr.io/{PROJECT_ID}/cc:" + tag_base + "-" + arch: "//cc:" + label + "_" + user + "_" + arch + "_debian9"
+    for arch in ARCHITECTURES
+    for (tag_base, label, user) in [
+        ("latest", "cc", "root"),
+        ("nonroot", "cc", "nonroot"),
+        ("debug", "debug", "root"),
+        ("debug-nonroot", "debug", "nonroot"),
+    ]
+})
+
+CC.update({
+    "gcr.io/{PROJECT_ID}/cc-" + distro + ":" + tag_base + "-" + arch: "//cc:" + label + "_" + user + "_" + arch + "_" + distro
+    for arch in ARCHITECTURES
+    for (tag_base, label, user) in [
+        ("latest", "cc", "root"),
+        ("nonroot", "cc", "nonroot"),
+        ("debug", "debug", "root"),
+        ("debug-nonroot", "debug", "nonroot"),
+    ]
+    for distro in DISTROS
+})
+
+PYTHON3 = {
+    "gcr.io/{PROJECT_ID}/python3:{COMMIT_SHA}": "//experimental/python3:python3_root_amd64_debian9",
+    "gcr.io/{PROJECT_ID}/python3-debian9:{COMMIT_SHA}": "//experimental/python3:python3_root_amd64_debian9",
+    "gcr.io/{PROJECT_ID}/python3-debian10:{COMMIT_SHA}": "//experimental/python3:python3_root_amd64_debian10",
+}
+
+PYTHON3.update({
+    "gcr.io/{PROJECT_ID}/python3:" + tag_base + "-" + arch: "//experimental/python3:" + label + "_" + user + "_" + arch + "_debian9"
+    for arch in BASE_ARCHITECTURES
+    for (tag_base, label, user) in [
+        ("latest", "python3", "root"),
+        ("nonroot", "python3", "nonroot"),
+        ("debug", "debug", "root"),
+        ("debug-nonroot", "debug", "nonroot"),
+    ]
+})
+
+PYTHON3.update({
+    "gcr.io/{PROJECT_ID}/python3-" + distro + ":" + tag_base + "-" + arch: "//experimental/python3:" + label + "_" + user + "_" + arch + "_" + distro
+    for arch in BASE_ARCHITECTURES
```

Comment:
Image tags in `python3-debian{9|10}` (as opposed to `python3`) were all amd64 images and not manifest lists. This code now pushes multiple architectures, which is fine. But probably you'll want to add `docker create manifest` in `cloudbuild_docker.sh`. Also worth updating the PR title to also add "convert `python{3|2.7}-debian{9|10}` multi-arch" for easy reference in the future.

Reply:
I made these use the same helper function for both python versions with the shorter architecture list.